tests/sql/test_sess_sql.py

```diff
@@ -0,0 +1,109 @@
+import pyarrow as pa
+import pytest
+
+from daft import Catalog, DataFrame, Session
+
+
+def assert_eq(actual: DataFrame, expect: DataFrame):
+    assert actual.to_pydict() == expect.to_pydict()
+
+
+def _create_catalog(name: str, tmpdir: str):
+    from pyiceberg.catalog.sql import SqlCatalog
+
+    catalog = SqlCatalog(
+        name,
+        **{
+            "uri": f"sqlite:///{tmpdir}/pytest_sql_{name}.db",
+            "warehouse": f"file://{tmpdir}",
+        },
+    )
+    # using naming convention "tbl_<catalog>_<namespace #><table #>"
+    # which let's us know where the table is coming from with only its name
+    catalog.create_namespace("ns_1")
+    catalog.create_namespace("ns_2")
+    catalog.create_table(f"ns_1.tbl_{name}_11", pa.schema([]))
+    catalog.create_table(f"ns_1.tbl_{name}_12", pa.schema([]))
+    catalog.create_table(f"ns_2.tbl_{name}_21", pa.schema([]))
+    catalog.create_table(f"ns_2.tbl_{name}_22", pa.schema([]))
+    return Catalog.from_iceberg(catalog)
+
+
+@pytest.fixture()
```

Comment:
nit: you can scope this to the session instead of recreating it every timi

```py
@pytest.fixture(scope="session")
```

Reply:
Ah I wasn't able to do so because of tmpdir scope, but looks like there are some workarounds! https://stackoverflow.com/questions/25525202/py-test-temporary-folder-for-the-session-scope